### PR TITLE
allow update vacation request status if admin

### PR DESCRIPTION
**Pull request by JuusoMRusanen**

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/timebank/api/impl/VacationRequestStatusApi.kt [hunk 1]
```diff
@@ -56,7 +56,7 @@ class VacationRequestStatusApi: VacationRequestStatusApi, AbstractApi() {
         val userId = loggedUserId ?: return createUnauthorized("Invalid token!")
         val existingStatus = vacationRequestStatusController.findVacationRequestStatus(statusId) ?: return createNotFound("Vacation request status not found")
 
-        if (existingStatus.createdBy != userId) return createForbidden("You can only edit your own statuses")
+        if (existingStatus.createdBy != userId && !isAdmin()) return createForbidden("You can only edit your own statuses")
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/timebank/api/impl/VacationRequestStatusApi.kt:59]: The error response message seems a bit odd.
Also this seems to be entirely missing integration tests for this new feature?

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changeLog.xml [hunk 2]
```diff
@@ -125,7 +125,7 @@
                 <constraints primaryKey="true" nullable="false"/>
             </column>
             <column name="vacationrequest_id" type="binary(16)">
-                <constraints foreignKeyName="FK_vacationrequest_id" referencedTableName="vacationRequest" referencedColumnNames="id"/>
+                <constraints foreignKeyName="FK_vacationrequest_id" referencedTableName="VacationRequest" referencedColumnNames="id"/>
```
Comment (villejuutila) [inline src/main/resources/db/changeLog.xml:128]: Revert this change. Migrations that are already applied in any real database should not be touched. Instead, configure your development database to run in case insensitive mode.

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestStatusTest.kt [hunk 3]
```diff
@@ -142,6 +142,46 @@ class VacationRequestStatusTest: AbstractTest() {
...
+    @Test
+    fun testUpdateVacationRequestStatusPermissions() {
+        createTestBuilder().use { testBuilder ->
+            val request = testBuilder.manager.vacationRequests.createVacationRequest(testVacationRequest)
+            val status = testBuilder.manager.vacationRequestStatus.createVacationRequestStatus(
+                requestId = request.id!!,
+                VacationRequestStatus(
+                    vacationRequestId = request.id,
+                    status = VacationRequestStatuses.PENDING,
+                    message = "Olen manager ja loin statuksen",
```
Comment (villejuutila) [inline src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestStatusTest.kt:157]: Keep these strings in english so everyone can understand :)
Applies to this whole file.

**Review by villejuutila — APPROVED** [review]
Approved with few comments.

File: src/main/resources/db/changeLog.xml (hunk 2 above)
Comment (JuusoMRusanen) [reply src/main/resources/db/changeLog.xml:128]: Reverting this change. Will have to find a way for the case insensitive mode.

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestStatusTest.kt (hunk 3 above)
Comment (JuusoMRusanen) [reply src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestStatusTest.kt]: Translated the strings in the file to English.
